### PR TITLE
[msbuild] Unify and relocate shared MSBuild overriden tasks

### PR DESCRIPTION
**Pull request by kzu**

### CODE REVIEW

**Comment by kzu** [comment]
@jstedfast @joj tech debt to improve the location and usage of shared MSBuild overriden tasks across iOS and XM targets

**Comment by xamarin-release-manager** [comment]
Build failure

**Comment by jstedfast** [comment]
This seems reasonable

File: msbuild/Xamarin.MacDev.Tasks.Core/MsBuildTasks/MoveTaskBase.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Microsoft.Build.Tasks
+{
+	public abstract class MoveTaskBase : Move
+	{
+        public string SessionId { get; set; }
+    }
+}
```
Comment (jstedfast) [inline msbuild/Xamarin.MacDev.Tasks.Core/MsBuildTasks/MoveTaskBase.cs]: This seem to have weird formatting

**Comment by kzu** [comment]
As soon as this is merged, we can update the XVS side to match. 

Thanks @jstedfast!

File: msbuild/Xamarin.MacDev.Tasks/MsBuildTasks/Move.cs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+﻿namespace Microsoft.Build.Tasks
+{
+    public class Move : MoveTaskBase
+	{
+	}
+}
```
Comment (jstedfast) [inline msbuild/Xamarin.MacDev.Tasks/MsBuildTasks/Move.cs]: More weird formatting

File: msbuild/Xamarin.MacDev.Tasks.Core/MsBuildTasks/MoveTaskBase.cs (hunk 1 above)
Comment (kzu) [inline msbuild/Xamarin.MacDev.Tasks.Core/MsBuildTasks/MoveTaskBase.cs]: Yep, fixed on the rebase after merging .editorconfig :)

**Comment by xamarin-release-manager** [comment]
Build failure

**Comment by kzu** [comment]
Really? Move does not exist in xbuild? :(

**Comment by kzu** [comment]
Conditionally added the legacy MoveTaskBase only when using xbuild to build (will automatically "just work" with MSBuild when we switch to it).

**Comment by xamarin-release-manager** [comment]
Build failure

**Comment by xamarin-release-manager** [comment]
Build success

**Comment by xamarin-release-manager** [comment]
Build success

**Comment by kzu** [comment]
ok @jstedfast, finally ready for merging! :+1:

**Pull request merged**